<launch>

    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find sopvo)/rviz/sopvo_t265_live.rviz" />
    <param name="/yamlconfigfile" type="string" value="$(find sopvo)/config/t265.yaml"/>
    <param name="/voparamfilepath" type="string" value="$(find sopvo)/config/vo_params_t265.yaml"/>
    <param name="/frameID" type="string" value="camera_odom_frame"/>
    <!--<param name="/output_file_path" type="string" value="$(find sopvo)/results/t265.txt" />-->
    <node pkg="rosbag" type="play" name="rosbag" args="$(find sopvo)/bag/t265.bag -r 0.5"/>
    <arg name="node_start_delay" default="0.0" />
    <param name="/voc"             type="string" value="$(find sopvo)/voc/voc_orb.dbow3"/>
    <!-- Manager -->
    <node pkg="nodelet" type="nodelet" name="sopvo_nodelet_manager"  args="manager" output="screen"
    launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' "/>
    <!-- TrackingNode, stereo vision only -->
    <node pkg="nodelet" type="nodelet" args="load sopvo/TrackingNodeletClass sopvo_nodelet_manager"
        name="TrackingNodeletClass_loader" output="screen"
        launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
        <param name="num_worker_threads" type="int" value="3" />
        <remap from="/vo/image0"   to="/camera/fisheye1/image_raw"/>
        <remap from="/vo/image1"   to="/camera/fisheye2/image_raw"/>
    </node>

</launch>
